OpenRA.Mods.Common/Pathfinder/PathSearch.cs

```diff
@@ -76,30 +80,101 @@ public static IPathSearch FromPoints(World world, Locomotor locomotor, Actor sel
+		/// Default: Diagonal distance heuristic. More information:
+		/// http://theory.stanford.edu/~amitp/GameProgramming/Heuristics.html
+		/// Layers are ignored and incur no additional cost.
+		/// </summary>
+		/// <param name="locomotor">Locomotor used to provide terrain costs.</param>
+		/// <param name="destination">The cell for which costs are to be given by the estimation function.</param>
+		/// <returns>A delegate that calculates the cost estimation between the <paramref name="destination"/> and the given cell.</returns>
+		public static Func<CPos, int> DefaultCostEstimator(Locomotor locomotor, CPos destination)
 		{
-			var cost = heuristic(location);
-			Graph[location] = new CellInfo(CellStatus.Open, 0, cost, location);
-			var connection = new GraphConnection(location, cost);
-			OpenQueue.Add(connection);
-			StartPoints.Add(connection);
+			var estimator = DefaultCostEstimator(locomotor);
+			return here => estimator(here, destination);
 		}
 
 		/// <summary>
-		/// This function analyzes the neighbors of the most promising node in the Pathfinding graph
+		/// Default: Diagonal distance heuristic. More information:
+		/// http://theory.stanford.edu/~amitp/GameProgramming/Heuristics.html
+		/// Layers are ignored and incur no additional cost.
+		/// </summary>
+		/// <param name="locomotor">Locomotor used to provide terrain costs.</param>
+		/// <returns>A delegate that calculates the cost estimation between the given cells.</returns>
+		public static Func<CPos, CPos, int> DefaultCostEstimator(Locomotor locomotor)
+		{
+			// Determine the minimum possible cost for moving horizontally between cells based on terrain speeds.
+			// The minimum possible cost diagonally is then Sqrt(2) times more costly.
+			var cellCost = locomotor.Info.TerrainSpeeds.Values.Min(ti => ti.Cost);
+			var diagonalCellCost = Exts.MultiplyBySqrtTwo(cellCost);
+			return (here, destination) =>
+			{
+				var diag = Math.Min(Math.Abs(here.X - destination.X), Math.Abs(here.Y - destination.Y));
+				var straight = Math.Abs(here.X - destination.X) + Math.Abs(here.Y - destination.Y);
+
+				// According to the information link, this is the shape of the function.
+				// We just extract factors to simplify.
+				// Possible simplification: var h = Constants.CellCost * (straight + (Constants.Sqrt2 - 2) * diag);
+				return cellCost * straight + (diagonalCellCost - 2 * cellCost) * diag;
+			};
+		}
+
+		public IGraph Graph { get; }
+		readonly Func<CPos, int> heuristic;
+		readonly int heuristicWeightPercentage;
+		public Func<CPos, bool> TargetPredicate { get; set; }
+		readonly IPriorityQueue<GraphConnection> openQueue;
+
+		/// <summary>
+		/// Initialize a new search.
+		/// </summary>
+		/// <param name="graph">Graph over which the search is conducted.</param>
+		/// <param name="heuristic">Provides an estimation of the distance between the given cell and the target.</param>
+		/// <param name="heuristicWeightPercentage">
+		/// Defaults to 100.
```

Comment:
It looks to me like this is defaulting to the 125 const?

Reply:
Have updated the comments to remove the "default" wording since this constructor doesn't even apply defaults, that's moved to the callers.